File: C/RPS/rps.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
void rps(int n)
{
    srand(time(0));
    for (int i = 0; i < n; i++)
    {
        int computer = rand() % 3;
        int player;
        printf("\n Enter your choice (0-Rock, 1-Paper, 2-Scissors): ");
        scanf("%d", &player);

        if (computer == player)
        {
            printf("Both Chose the same... It's a tie!! \n");
        }
        else if (computer == 0 && player == 1)
        {
            printf("Computer chose Rock... Player Wins! \n");
        }
        else if (computer == 0 && player == 2)
        {
            printf("Computer chose Rock... Computer Wins! \n");
        }
        else if (computer == 1 && player == 2)
        {
            printf("Computer chose Paper... Player Wins! \n");
        }
        else if (computer == 1 && player == 0)
        {
            printf("Computer chose Paper... Computer Wins! \n");
        }
        else if (computer == 2 && player == 0)
        {
            printf("Computer chose Scissors... Player Wins! \n");
        }
        else if (computer == 2 && player == 1)
        {
            printf("Computer chose Scissors... Computer Wins! \n");
        }
        else{
            printf("Invalid choice... Please try again! \n");
            i--;
            continue;
        }
    }
}
int main()
{
    rps(3);
    return 0;
}

File: Python/RPS.py
import random as rand
import time

rock = ['r', 'R', 'Rock', 'rock']
paper = ['p', 'P', 'Paper', 'paper']
scissors = ['s', 'S', 'Scissors', 'scissors', 'Scissor', 'scissor']


def RPS():
    player_score = 0
    comp_score = 0
    game_number = 1


    total_games = int(input('How Many Games Would You Like To Play? '))
    time.sleep(2)
    
    while game_number <= total_games:
        print()
        print('*'*30)
        print('Game Number:', game_number)
        print('*'*30)
        print()
        time.sleep(2)
        player_choice = input('Enter Your Choice (Rock/Paper/Scissors) OR (r/p/s): ').lower()
        comp_choice = rand.choice(['Rock', 'Paper', 'Scissors'])
        print()
        time.sleep(2)
        if player_choice in rock and comp_choice in rock:
            print('''Both chose Rock...It's a Tie!''')
        elif player_choice in paper and comp_choice in paper:
            print('''Both chose Paper...It's a Tie!''')
        elif player_choice in scissors and comp_choice in scissors:
            print('''Both chose Scissors...It's a Tie!''')
        elif player_choice in rock and comp_choice in paper:
            print('Computer chooses Paper...You lose!!')
            comp_score += 1
        elif player_choice in rock and comp_choice in scissors:
            print('Computer chooses Scissors...You Win!!')
            player_score += 1
        elif player_choice in paper and comp_choice in rock:
            print('Computer chooses Rock...You Win!!')
            player_score += 1
        elif player_choice in paper and comp_choice in scissors:
            print('Computer chooses Scissors...You Lose!!')
            comp_score += 1
        elif player_choice in scissors and comp_choice in rock:
            print('Computer chooses Rock...You Lose!!')
            comp_score += 1
        elif player_choice in scissors and comp_choice in paper:
            print('Computer chooses Paper...You Win!!')
            player_score += 1
        else:
            print('Invalid Choice! Game Discarded!')
        game_number += 1
        time.sleep(2)

    print()
    print('*'*30)
    print('FINAL SCORES:')
    time.sleep(2)
    print('Player Score:', player_score)
    time.sleep(2)
    print('Computer Score:', comp_score)
    print('*'*30)
    print()
    time.sleep(2)
    print('Thank You for playing')

RPS()


